routellm/routers/matrix_factorization/model.py

```diff
@@ -74,18 +74,46 @@
 class MFModel(torch.nn.Module, PyTorchModelHubMixin):
     def __init__(
         self,
-        dim,
-        num_models,
-        text_dim,
-        num_classes,
-        use_proj,
+        dim=128,
+        num_models=64,
+        text_dim=768,
+        num_classes=1,
+        use_proj=True,
+        collapse_linear=False,
+        embedding_model="all-mpnet-base-v2",
```

Comment:
I think it would be better if we didn't set any default args here and specify default args only in `routers.py` so that it's easier to keep track of router configs.